src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.MultiSegment.cs

```diff
@@ -0,0 +1,2408 @@
+
+            if (_consumed >= (uint)_buffer.Length)
+            {
+                if (!_isNotPrimitive)
+                {
+                    return true;
+                }
+                if (IsLastSpan || !GetNextSpan())
+                {
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedEndOfDigitNotFound, _buffer[_consumed - 1]);
+                }
+            }
+
+            // TODO: https://github.com/dotnet/corefx/issues/33294
+            if (JsonConstants.Delimiters.IndexOf(_buffer[_consumed]) < 0)
+            {
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedEndOfDigitNotFound, _buffer[_consumed]);
+            }
+            return true;
+        }
+
+        private bool ConsumePropertyNameMultiSegment()
+        {
+            if (!ConsumeStringMultiSegment())
+            {
+                return false;
+            }
+
+            if (!HasMoreDataMultiSegment(ExceptionResource.ExpectedValueAfterPropertyNameNotFound))
+            {
+                return false;
+            }
+
+            byte first = _buffer[_consumed];
+
+            // This check is done as an optimization to avoid calling SkipWhiteSpace when not necessary.
+            // We do not validate if 'first' is an invalid JSON byte here (such as control characters).
+            // Those cases are captured below where we only accept ':'.
+            if (first <= JsonConstants.Space)
+            {
+                SkipWhiteSpaceMultiSegment();
+                if (!HasMoreDataMultiSegment(ExceptionResource.ExpectedValueAfterPropertyNameNotFound))
+                {
+                    return false;
+                }
+                first = _buffer[_consumed];
+            }
+
+            // The next character must be a key / value seperator. Validate and skip.
+            if (first != JsonConstants.KeyValueSeperator)
+            {
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedSeparatorAfterPropertyNameNotFound, first);
+            }
+
+            _consumed++;
+            _bytePositionInLine++;
+            _tokenType = JsonTokenType.PropertyName;
+            return true;
+        }
+
+        private bool ConsumeStringMultiSegment()
+        {
+            Debug.Assert(_buffer.Length >= _consumed + 1);
+            Debug.Assert(_buffer[_consumed] == JsonConstants.Quote);
+
+            // Create local copy to avoid bounds checks.
+            ReadOnlySpan<byte> localBuffer = _buffer.Slice(_consumed + 1);
+
+            // Vectorized search for either quote, backslash, or any control character.
+            // If the first found byte is a quote, we have reached an end of string, and
+            // can avoid validation.
+            // Otherwise, in the uncommon case, iterate one character at a time and validate.
+            int idx = localBuffer.IndexOfQuoteOrAnyControlOrBaskSlash();
```

Comment:
I know it's an existing problem, but I've seen this now and can't unsee it: The method says Bask Slash instead of Back Slash.  And now I'm imagining slashes lying on deck chairs, basking in the sun.  (I'm concerned someone's going to lay on them, since they look like the deck chairs themselves)

Reply:
Lol.  Literally.

Reply:
> And now I'm imagining slashes lying on deck chairs, basking in the sun. (I'm concerned someone's going to lay on them, since they look like the deck chairs themselves)

But the question remains... what's their index?